Create DIVD-2024-00032.md (#805)
DIVD-NL/web-csirt: _cases/2024/DIVD-2024-00032.md

---
layout: case
title: "Unauthenticated Remote Code Execution (RCE) vulnerability in Geoserver"
author: Koen Schagen
lead: Alwin Warringa
excerpt: "Geoserver has a Remote Code Execution (RCE) vulnerability in evaluating property name expressions"
researchers:
- Alwin Warringa
- Koen Schagen
- Stan Plasmeijer
cves:

> - CVE-2024-36401

Reviewer: ```suggestion
- CVE-2024-36401
- CVE-2024-36404
```

product:
- Geoserver
versions:

> - 2.24.4 and earlier

Reviewer: ```suggestion
- GeoServer version 2.24.0 or later, but earlier than version 2.24.4.
```

- 2.25.2 and earlier
- 2.23.6 and earlier
recommendation: "Check for the patched versions and get those installed"
workaround: "Yes"
patch_status: Released
status : Open
start: 2024-07-03
end:
timeline:
- start: 2024-07-03
  end:
  event: "DIVD starts researching the vulnerability."
- start: 2024-07-03
  end:
  event: "DIVD finds fingerprint, preparing to scan."
- start: 2024-07-04
  end:
  event: "DIVD starts scanning the internet for vulnerable instances."
# ips: 0

---

## Summary

> Geoserver has a Remote Code Execution (RCE) vulnerability in evaluating property name expressions that allows an unauthenticated attacker get full access to the system.

Reviewer: ```suggestion
GeoServer and GeoTools have issued a security announcement and fixed an XPath expression injection vulnerability. This vulnerability results in unauthenticated Remote Code Execution (RCE), allowing an attacker to gain full access to the system without authentication.
```

## Recommendations

> Geoserver released serveral patched versions (2.24.4, 2.25.2, 2.23.6) Please update to one of those versions (or better) as soon as possible.

Reviewer: ```suggestion
GeoServer released serveral patched versions (2.24.4, 2.25.2 and 2.23.6). Please update to one of those versions (or better) as soon as possible.

GeoTools released serveral patched versions (29.6, 30.4 and 31.2). Please update to one of those versions (or better) as soon as possible.
```

> 

Reviewer: ```suggestion

## Mitigation

If relevant users cannot install updates temporarily, the following measures can be taken for temporary relief: Deleting the `gt-complex-x.y.jar` file in GeoServer (x.y is the version of GeoTools, such as `gt-complex-31.1.jar` in GeoServer 2.25.1) will remove vulnerable code from GeoServer, but may compromise some GeoServer functionality. When a gt-complex module is required by an extension in use, it may cause the GeoServer deployment to fail.
```
